packages/core/src/collider/Collider.ts

```diff
@@ -20,4 +27,43 @@ export class Collider extends Component {
   _onDisable(): void {
     this.scene.findFeature(ColliderFeature).detachCollider(this);
   }
+
+  /**
+   * Calculate the raycasthit in world space.
+   * @param ray - The ray
+   * @param distance - The distance
+   * @param outHit - The raycasthit
+   * @private
+   */
+  _updateHitResult(ray: Ray, distance: number, outHit: HitResult, origin: Vector3, isWorldRay: boolean = false) {
```

Comment:
Maybe this func can should delete @priavte and add protected: `protected  _updateHitResult()`